packages/api-specification/api/windows.ts

```diff
@@ -0,0 +1,313 @@
+
+    /**
+     * Receive notifications when a new window is opened
+     * @returns Function Execute the function to unsubscribe
+     */
+    onWindowOpened(callback: (window: Window) => void): UnsubscribeFunction;
+
+    /**
+     * Receive notifications when a window was closed
+     * @returns Function Execute the function to unsubscribe
+     */
+    onWindowClosed(callback: (window: Window) => void): UnsubscribeFunction;
+
+    /**
+     * Subscribe for bounds changed event for all windows
+     * @returns Function Execute the function to unsubscribe
+     */
+    onWindowBoundsChanged(callback: (window: Window, bounds: Bounds) => void): UnsubscribeFunction;
+}
+
+/** Window options used to open a new window **/
+export interface WindowOptions {
+    /**
+     * Default window title.
+     */
+    name: string;
+
+    /**
+     * URL that this window loads.
+     */
+    url: string;
+
+    /**
+     * Window initial bounds. Can contain partial bounds (e.g width only) and be
+     * combined with center
+     */
+    bounds?: Bounds;
+
+    /**
+     * Window’s maximum width.
+     */
+    maxWidth?: number;
+
+    /**
+     * Window’s minimum width.
+     */
+    minWidth?: number;
+
+    /**
+     * Window’s maximum height.
+     */
+    maxHeight?: number;
+
+    /**
+     * Window’s minimum height.
+     */
+    minHeight?: number;
+
+    /**
+     * Whether the window should always stay on top of other windows. Default is false.
+     */
+    alwaysOnTop?: boolean;
+
+    /**
+     * Window’s background color as Hexadecimal value.
+     */
+    backgroundColor?: string;
+
+    /**
+     * Whether the window is a child of the current window. Default is false.
+     */
+    child?: boolean;
+
+    /**
+     * Show window in the center of the screen.
+     */
+    center?: boolean;
+
+    /**
+     * If false, creates a frameless window. Default is true.
+     */
+    frame?: boolean;
+
+    /**
+     * Whether window should have a shadow. This is only implemented on macOS. Default is true.
+     */
+    hasShadow?: boolean;
+
+    /**
+     * Whether window is maximizable. Default is true.
+     */
+    maximizable?: boolean;
+
+    /**
+     * Whether window is minimizable. Default is true.
+     */
+    minimizable?: boolean;
+
+    /**
+     * Whether window is resizable. Default is true.
+     */
+    resizable?: boolean
+
+    /**
+     * Whether window should be shown when created. Default is true.
+     */
+    show?: boolean;
+
+    /**
+     * Whether to show the window in taskbar. Default is false.
+     */
+    skipTaskbar?: boolean;
+
+    /**
+     * Makes the window transparent. Default is false.
+     */
+    transparent?: boolean;
+}
+
+/**
+ * A window running in the container.
+ * It implements EventEmitter which means users can subscribe for events using on, once , addEventListener etc.
+ * The list of events is in WindowsEvents structure;
+ */
+export interface Window extends EventEmitter {

```

Comment:
using window.open (with sandbox on) in electron like this works just fine, the child can use the window.opener.  we have this working currently in electron.  when window.open url has the same domain, then it runs in the same process and the child can access the parent.  

but yes if in diff process then window.opener won't work without some extra method to proxy between process across ipc.